safe/gui/tools/batch/batch_dialog.py

```diff
@@ -500,14 +524,52 @@ def run_task(self, task_item, status_item, count=0, index=''):
 
                 # Load impact layer into QGIS
                 qgis_layer = read_impact_layer(impact_layer)
-                QgsMapLayerRegistry.instance().addMapLayer(
-                    qgis_layer, addToLegend=False)
-
+                impact_layer_source = qgis_layer.source()
+                reg.instance().addMapLayer(qgis_layer, addToLegend=False)
+
+                # identify impact layer in map canvas from impact layer source
+                leg_impact_layer = self.identify_layer(impact_layer_source)
+                # clone impact layer in Layer Panel
+                if qgis_layer.type() == QgsMapLayer.VectorLayer:
+                    cloned_layer = QgsVectorLayer(
+                        leg_impact_layer.source(),
+                        leg_impact_layer.name(),
+                        leg_impact_layer.providerType())
+                elif qgis_layer.type() == QgsMapLayer.RasterLayer:
+                    cloned_layer = QgsRasterLayer(
+                        leg_impact_layer.source(),
+                        leg_impact_layer.name(),
+                        leg_impact_layer.providerType())
+                else:
+                    raise Exception('layer source is failed to be recognized')
+                # turn off layer visibility just in case layer removing
+                # is not working so that it won't be shown in the next scenario
+                legend_interface = self.iface.legendInterface()
+                legend_interface.setLayerVisible(leg_impact_layer, False)
+                # remove original layer created from dock, sometimes this
+                # code is failed leaving original file in layer panel
+                reg.instance().removeMapLayer(leg_impact_layer)
+                # add cloned layer to the layer group
+                reg.instance().addMapLayer(cloned_layer, False)
+                self.layer_group.insertLayer(0, cloned_layer)
+                # turn off exposure layer visibility
+                exposure_layer = self.identify_layer(self.exposure_source)
+                legend_interface.setLayerVisible(exposure_layer, False)
+
+                # print layer list. somehow this is the only way to update
+                # the map canvas. still need improvement
+                layers = self.iface.mapCanvas().layers()
+                for layer in layers:
+                    print layer.name()
                 # noinspection PyBroadException
+                # status_item.setText(self.tr('Analysis Ok'))
+                # self.create_pdf(title, path, cloned_layer, count, index)
+                # status_item.setText(self.tr('Report OK'))
```

Comment:
sorry @Gustry , I forgot pushing the changes to the branch a few days ago, and when I check it right now I just realised it.
I already pushed it now.